projects/iio_demo_freeRTOS/src/platform/maxim/main.c

```diff
@@ -0,0 +1,143 @@
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include "platform_includes.h"
+#include "common_data.h"
+
+#ifdef IIO_EXAMPLE
+#include "iio_example.h"
+#endif
+
+/**
+ * @brief IIOD example task
+ * @return Returns the error code of iio_example_main function
+*/
+int iiodTask()
+{
+	return iio_example_main();
+}
+
+/**
+ * @brief LED blinking task for multithreading example
+ * @return Returns error code
+*/
+int blinkingTask()
+{
+	int ret = 0;
+	struct no_os_gpio_desc *led_pin;
+	ret = no_os_gpio_get(&led_pin, &gpio_ip);
+	if (ret)
+		return ret;
+
+	ret = no_os_gpio_direction_output(led_pin, NO_OS_GPIO_LOW);
+	if (ret)
+		return ret;
+
+	bool led_on = true;
+	while (1) {
+		if (!led_on) {
+			ret = no_os_gpio_set_value(led_pin, NO_OS_GPIO_HIGH);
+			if (ret)
+				break;
+			led_on=true;
+			no_os_mdelay(100);
+		}
+		if (led_on) {
+			ret = no_os_gpio_set_value(led_pin, NO_OS_GPIO_LOW);
+			if (ret)
+				break;
+			led_on=false;
+			no_os_mdelay(100);
+		}
+	}
+	return ret;
+}
+
+/**
+ * @brief Creating iio example with blinking led task
+ * @return  Returns error code
+ *
+*/
+int create_tasks(void)
+{
+	int ret;
+	for (int i = 0; i < 0x00FFFF; i++) {}
+
+	ret = xTaskCreate(iiodTask, (const char *)"iiodTask", configIIO_APP_STACK_SIZE,
+			  NULL, tskIDLE_PRIORITY + 1, NULL);
+	if (ret != pdPASS) {
+		printf("xTaskCreate() failed to create iiodTask task.\n");
+		return ret;
+	}
+	ret = xTaskCreate(blinkingTask, (const char *)"blinkingTask",
+			  configMINIMAL_STACK_SIZE, NULL, tskIDLE_PRIORITY + 1, NULL);
+	if (ret != pdPASS) {
+		printf("xTaskCreate() failed to create blinkingTask task.\n");
+		return ret;
```

Comment:
is there any dynamic allocation happening when xTaskCreate is called? if so, before returning make sure that memory is freed

Reply:
In case the task creation is not successful, there should not be any allocated memory. There is a xTaskDelete function for deleting tasks, I have added that as a safety step.